### PR TITLE
chore: dynamic shape support for pdist ops

### PR DESCRIPTION
**Pull request by keehyuna**
# Description

np.triu_indices(dim0, k=1) is implemented by index calculation in loop layer when dim0 is dynamic shape

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py [hunk 1]
```diff
@@ -417,20 +421,21 @@ def pdist(
 ) -> Union[TRTTensor, Sequence[TRTTensor]]:
     shape = input.shape
     # Extend input from shape [N, D] to [N, 1, D]
-    extend_input = impl.shuffle.reshape(
+    extend_input = impl.unsqueeze.unsqueeze(
```
Comment (keehyuna) [inline py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:424]: same functionality. it's changed to handle when all shapes are dynamic shape

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py [hunk 2]
```diff
@@ -482,8 +487,207 @@ def pdist(
...
+    one_layer = ctx.net.add_constant(
+        shape=(), weights=trt.Weights(np.array([1], dtype=np.int32))
+    )
+    set_layer_name(one_layer, target, f"{name}_one")
+    constant_1 = one_layer.get_output(0)
+    two_layer = ctx.net.add_constant(
+        shape=(), weights=trt.Weights(np.array([2], dtype=np.int32))
+    )
+    set_layer_name(two_layer, target, f"{name}_two")
+    constant_2 = two_layer.get_output(0)
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:554]: Can you directly call `create_constant` in converter_utils.py for simplification?

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py [hunk 3]
```diff
@@ -482,8 +487,207 @@ def pdist(
...
+    loop_output_y.set_input(1, num_loop)
+
+    # Cat two N tensor into 2 x N. [0, 0, 0], [1, 2, 3] -> [[0, 0, 0], [1, 2, 3]]
+    x_index = ctx.net.add_shuffle(loop_output_x.get_output(0))
+    set_layer_name(x_index, target, f"{name}_x_index", source_ir)
+    x_index.reshape_dims = (1, -1)
+
+    y_index = ctx.net.add_shuffle(loop_output_y.get_output(0))
+    set_layer_name(x_index, target, f"{name}_y_index", source_ir)
+    y_index.reshape_dims = (1, -1)
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:673]: Can you directly call `reshape`?

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py [hunk 4]
```diff
@@ -482,8 +487,207 @@ def pdist(
...
+        f"{name}_x_y_tensor",
+        [x_index.get_output(0), y_index.get_output(0)],
+        0,
+    )
+
+    # Reshape 2 x N output to N x 2. [[0, 0, 0], [1, 2, 3]] -> [[0, 1], [0, 2], [0, 3]]
+    indices_tensor = ctx.net.add_shuffle(x_y_tensor)
+    set_layer_name(indices_tensor, target, f"{name}_indices_tensor", source_ir)
+    indices_tensor.first_transpose = trt.Permutation([1, 0])
+    indices_tensor.reshape_dims = (0, 0)
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:688]: What's the meaning of `indices_tensor.reshape_dims = (0, 0)`?

**Review by zewenli98 — CHANGES_REQUESTED** [review]

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py (hunk 2 above)
Comment (keehyuna) [reply py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py]: Thanks, It also helped to find wrong import from torch_tensorrt.fx.converters.converter_utils.

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py (hunk 4 above)
Comment (keehyuna) [reply py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py]: According to doc, 0s are resolved by aligning the most significant dimensions of input.
https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Graph/Layers.html#ishufflelayer
transposed shape from indices_tensor.first_transpose is N x 2. reshape_dims = (0, 0) will keep use it.
Other valid reshape_dims are (-1, 2),  (-1, 0), (0, -1). I changed to (-1, 2) for better reading.

**Review by zewenli98 — APPROVED** [review]
Thanks for the explanations. LGTM

**Review by peri044 — APPROVED** [review]
LGTM

**Pull request merged**